### PR TITLE
Update .NET docs

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Updates to .NET guidance.  Merge after releasing https://github.com/getsentry/sentry-dotnet/pull/2319

### CODE REVIEW

File: src/platforms/dotnet/common/troubleshooting.mdx [hunk 1]
```diff
@@ -43,53 +43,14 @@ See [this GitHub issue](https://github.com/getsentry/sentry-dotnet/issues/258) f
...
-  ```
-
-  Note that with this approach, it is not advised to use `.Wait()` or `.Result`, as that would wrap unhandled exceptions in an `AggregateException`.
-
-See [this GitHub issue](https://github.com/getsentry/sentry-dotnet/issues/321) for more context.
+Starting with C# 7.1, a program's [`Main` method](https://docs.microsoft.com/dotnet/csharp/fundamentals/program-structure/main-command-line) can be declared either synchronously or asynchronously.
+Later, in C# 9, this was further enhanced by use of [top-level statements](https://learn.microsoft.com/dotnet/csharp/fundamentals/program-structure/top-level-statements), which can also be declared either synchronously or asynchronously
+
+When using either of these features with the Sentry .NET SDK, an unhandled exception may not have been captured if `SentrySdk.Init` was called with a `using` statement.
+Additionally, any associated transaction should have be marked as failed and sent to Sentry - but may not have been when using an async main.
```
Comment (lizokm) [inline src/platforms/dotnet/common/troubleshooting.mdx:50]:
```suggestion
If `SentrySdk.Init` is called with a `using` statement while using either of these features with the Sentry .NET SDK,   unhandled exceptions might not get captured. Additionally, if you're using an async main, associated transactions might not get marked as failed and sent to Sentry.
```

File: src/platforms/dotnet/common/troubleshooting.mdx [hunk 2]
```diff
@@ -43,53 +43,14 @@ See [this GitHub issue](https://github.com/getsentry/sentry-dotnet/issues/258) f
...
+
+These issues were fully resolved as of version 3.31.0 of the Sentry .NET SDK.  If you are using an older version, simply update to the newest release any you can safely use an async main.
+
+Also, it is no longer required to dispose the result of calling `SentrySdk.Init`.
```
Comment (lizokm) [inline src/platforms/dotnet/common/troubleshooting.mdx:54]:
```suggestion
These issues have been fully resolved as of version 3.31.0 of the Sentry .NET SDK. To safely use an async main, update your version to the newest release. It's no longer required to dispose of the result of calling `SentrySdk.Init`.
```

File: src/wizard/dotnet/index.md [hunk 3]
```diff
@@ -19,23 +19,34 @@ Install-Package Sentry -Version {{ packages.version('sentry.dotnet') }}
...
-}
+    // A Sentry Data Source Name (DSN) is required.
+    // See https://docs.sentry.io/product/sentry-basics/dsn-explainer/
+    // You can set it in the SENTRY_DSN environment variable, or you can set it in code here.
+    options.Dsn = "___PUBLIC_DSN___";
+
+    // When debug is enabled, the Sentry client will emit detailed debugging information to the console.
+    // This might be helpful, or might interfere with the normal operation of your application.
+    // We enable it here for demonstration purposes when first trying Sentry.
+    // You should not do this in your applications unless you are troubleshooting issues with Sentry.
```
Comment (lizokm) [inline src/wizard/dotnet/index.md:37]:
```suggestion
    // You shouldn't do this in your applications unless you're troubleshooting issues with Sentry.
```

File: src/wizard/dotnet/index.md [hunk 4]
```diff
@@ -19,23 +19,34 @@ Install-Package Sentry -Version {{ packages.version('sentry.dotnet') }}
...
+    options.Debug = true;
+
+    // This option is recommended, which enables Sentry's "Release Health" feature.
```
Comment (lizokm) [inline src/wizard/dotnet/index.md:40]:
```suggestion
    // This option is recommended. It enables Sentry's "Release Health" feature.
```

File: src/wizard/dotnet/index.md [hunk 5]
```diff
@@ -19,23 +19,34 @@ Install-Package Sentry -Version {{ packages.version('sentry.dotnet') }}
...
+    options.AutoSessionTracking = true;
+
+    // This option is recommended for client applications only. It ensures all threads use the same global scope.
+    // If you are writing a background service of any kind, you should remove this.
```
Comment (lizokm) [inline src/wizard/dotnet/index.md:44]:
```suggestion
    // If you're writing a background service of any kind, you should remove this.
```

**Review by lizokm — APPROVED** [review]
A few small language tweaks.

**Comment by mattjohnsonpint** [comment]
Thanks! Holding for a bit to align with the SDK release. Appreciate the feedback though!

**Comment by shanamatthews** [comment]
@mattjohnsonpint it looks like the SDK work was merged. Is this PR okay to merge once Liza's edits are incorporated?

**Pull request merged**